unittest/MultiVectorIndex.h

```diff
@@ -0,0 +1,141 @@
+#include "knowhere/index/vector_index/IndexBinaryIDMAP.h"
+#include "knowhere/index/vector_index/IndexBinaryIVF.h"
+#include "knowhere/index/vector_index/IndexHNSW.h"
+#include "knowhere/index/vector_index/IndexIDMAP.h"
+#include "knowhere/index/vector_index/IndexIVF.h"
+#include "knowhere/index/vector_index/IndexIVFPQ.h"
+#include "knowhere/index/vector_index/IndexIVFSQ.h"
+#ifdef KNOWHERE_WITH_DISKANN
+#include "knowhere/index/vector_index/IndexDiskANN.h"
+#include "knowhere/index/vector_index/IndexDiskANNConfig.h"
+#endif
+
+namespace knowhere {
+
+class MultiVectorIndex : public VecIndex {
+ public:
+    MultiVectorIndex(std::string type) {
+        std::transform(type.begin(), type.end(), type.begin(), toupper);
+        if (type == IndexEnum::INDEX_ANNOY) {
+            multithread_index_ptr = std::make_unique<knowhere::IndexAnnoy>();
+        } else if (type == IndexEnum::INDEX_FAISS_BIN_IDMAP) {
+            multithread_index_ptr = std::make_unique<knowhere::BinaryIDMAP>();
+        } else if (type == IndexEnum::INDEX_FAISS_BIN_IVFFLAT) {
+            multithread_index_ptr = std::make_unique<knowhere::BinaryIVF>();
+        } else if (type == IndexEnum::INDEX_FAISS_IDMAP) {
+            multithread_index_ptr = std::make_unique<knowhere::IDMAP>();
+        } else if (type == IndexEnum::INDEX_HNSW) {
+            multithread_index_ptr = std::make_unique<knowhere::IndexHNSW>();
+        } else if (type == IndexEnum::INDEX_FAISS_IVFFLAT) {
+            multithread_index_ptr = std::make_unique<knowhere::IVF>();
+        } else if (type == IndexEnum::INDEX_FAISS_IVFPQ) {
+            multithread_index_ptr = std::make_unique<knowhere::IVFPQ>();
+        } else if (type == IndexEnum::INDEX_FAISS_IVFSQ8) {
+            multithread_index_ptr = std::make_unique<knowhere::IVFSQ>();
+        } else {
+            KNOWHERE_THROW_FORMAT("Invalid index type %s", std::string(type).c_str());
+        }
+    }
+
+#ifdef KNOWHERE_WITH_DISKANN
+    MultiVectorIndex(std::string type, MetricType metric_type, std::shared_ptr<FileManager> file_manager) {
+        if (type == "diskann_f") {
+            multithread_index_ptr = std::make_unique<knowhere::IndexDiskANN<float>>(type, metric_type, file_manager);
+        } else if (type == "disann_ui8") {
+            multithread_index_ptr = std::make_unique<knowhere::IndexDiskANN<uint8_t>>(type, metric_type, file_manager);
+        } else if (type == "diskann_i8") {
+            multithread_index_ptr = std::make_unique<knowhere::IndexDiskANN<int8_t>>(type, metric_type, file_manager);
+        } else {
+            KNOWHERE_THROW_FORMAT("Invalid index type %s", std::string(type).c_str());
+        }
+    }
+#endif
+
+    std::vector<std::future<DatasetPtr>>&
+    GetFeatureStack() {
+        static std::vector<std::future<DatasetPtr>> stk;
+        return stk;
+    }
+
+    void
+    QueryAsync(const DatasetPtr& dataset, const Config& config, const faiss::BitsetView bitset) {
```

Comment:
This is a little bit weird to me:
1. If we want this class to extend `VecIndex`, then we need to use it as a `VecIndex` and don't give extra public API. 
2. If we need to support another group of API, we don't make the inheritance.

In our case, all APIs from `VecIndex` is useless. We can:
1. If swig support function param, we can simple rename this class to something has nothing to do with Index but async like `AsyncRequestEmitter`, and only keep `QueryASync`, `Sync`. `QueryAsync` will accept a function and do async call here.
2. If swig doesn't support that. We can rename it to something like `AsyncIndexProxy`, and remove all `VecIndex` APIs.